src/features/search/components/SearchListItem.web.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { CSSProperties } from 'react'
+import styled from 'styled-components/native'
+
+import { SearchListFooter } from 'features/search/components/SearchListFooter/SearchListFooter.web'
+import { SearchListHeader } from 'features/search/components/SearchListHeader/SearchListHeader'
+import { SearchListProps, SearchState } from 'features/search/types'
+import { AlgoliaVenue } from 'libs/algolia'
+import { Offer } from 'shared/offer/types'
+import { HorizontalOfferTile } from 'ui/components/tiles/HorizontalOfferTile'
+import { getSpacing } from 'ui/theme'
+
+export type RowData = {
+  items: (Record<string, never> | Offer)[]
```

Comment:
je trouve ça très bizarre un `Record<string, never>` pourquoi ne pas faire un `Set<string>` ?

Reply:
`Set<string>` ? Je ne connaissais pas. 
Donc ça marcherait ça ?
```ts
type RowItem = Set<string>
const test: RowItem = {}
```

Reply:
non, il faudrait faire un `new Set()` https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Set

`Record<string, never>` je le comprends comme : on a un objet (une map), avec des clés, et où on se fout des valeurs

une structure de données qui a des clés uniques sans valeur associée, ça m'évoque un `Set`

je ne comprends pas l'interet d'avoir une map si on ne s'intéresse pas aux valeurs

soit le `never` est incorrecte, soit la structure de donnée est inutilement complexe

Reply:
Ah tu n'as pas dû comprendre mon commentaire dans le code

```ts
      /**
       * This one in special since we manually add two empty objects.
       * The first one is used as a placeholder to place Header component.
       * The last one is used as a placeholder to place Footer component.
       *
       * This is necessary since the Row component (`SearchListItem.web`) is generic and we need to
       * guess what we want to draw.
       */
      items: [{}, ...hits.offers, {}],
```

En fait je ne veux ni de Set ni de rien. J'aurai pu mettre un string aussi. L'idée c'est juste de savoir qu'à l'index 0 je n'ai pas d'item, et à l'index -1 non plus. Alors pour concrétiser cette idée je mets des objets vides. Et pour que TypeScript soit OK avec ça je lui donne la définition d'un objet vide : `Record<string, never>`

Reply:
en effet, j'avais pas tout bien compris

dans ce cas, ce qui m'aurait aidé à comprendre, c'est d'avoir quelque chose comme

```ts
export const HeaderPlaceholder = {} as const
export const FooterPlaceholder = {} as const

export type RowData = {
  items: (typeof HeaderPlaceholder | Offer | typeof FooterPlaceholder)[]
```

```ts
      items: [HeaderPlaceholder, ...hits.offers, FooterPlaceholder],
```

Reply:
Ce qui aurait été beaucoup plus verbeux ! Mais oui je suis OK dans l'idée.

Reply:
un peu plus verbeux mais beaucoup plus explicite